src/main.rs

```diff
@@ -146,20 +162,103 @@ fn collect_information_and_write_to_file(out_path: String) {
         if let Some(message) = maybe_message {
             let result = format!("{} {}\n", emoji, message);
 
-            let mut f = File::create(out_path).unwrap();
+            let mut f = File::create(out_path.clone()).unwrap();
             f.write_all(result.as_bytes()).unwrap();
+
+            if launch_editor {
+                launch_default_editor(out_path);
+            }
         }
     }
 }
 
-fn main() {
-    if let Some(out_path) = env::args().nth(1) {
-        if out_path.ends_with(".git/COMMIT_EDITMSG") {
-            collect_information_and_write_to_file(out_path);
+#[derive(Debug)]
+struct ValidationError;
+
+impl fmt::Display for ValidationError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", "One or more commits has validation errors")
+    }
+}
+
+impl Error for ValidationError {}
+
+fn validate(refspecs: Vec<String>) -> Result<(), Box<dyn Error>> {
+    let repo_path = env::current_dir()?;
+    let messages = git::get_commit_messages(repo_path, refspecs)?;
+
+    let mut has_errors = false;
+    for message in messages {
+        let validation_errors = commit_rules::check_message_with_emoji(&message)
+            .filter_map(|result| {
+                if !result.pass { Some(format!("\t{}", result)) } else { None }
+            })
+            .collect::<Vec<_>>();
+        if validation_errors.len() > 0 {
+            has_errors = true;
+            let validation_result_text = validation_errors.join("\r\n");
+            let text = format!(
+                "\r\nCommit:\r\n\t{}\r\nValidation Errors:\r\n{}\r\n",
+                message,
+                validation_result_text,
+            );
+            println!("{}", text);
+        }
+    }
+    if has_errors {Err(Box::new(ValidationError{}))} else { Ok(()) }
+}
+
+#[derive(Debug)]
+enum OutPath {
+    EditMessage(PathBuf),
+    RebaseTodo(PathBuf),
+}
+
+impl FromStr for OutPath {
+    type Err = String;
+    fn from_str(raw: &str) -> Result<Self, Self::Err> {
+        let path = PathBuf::from(raw);
+        if path.ends_with(".git/COMMIT_EDITMSG") {
+            Ok(OutPath::EditMessage(path))
+        } else if path.ends_with(".git/rebase-merge/git-rebase-todo") {
+            Ok(OutPath::RebaseTodo(path))
         } else {
+            Err(String::from("Must end with one of .git/COMMIT_EDITMSG and .git/rebase-merge/git-rebase-todo"))
+        }
+    }
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(about = "Make your git logs beautiful and readable with the help of emojis 🎉")]
+struct Opt {
+    #[structopt(
+        help = "Path passed by Git where commit message should be written to. Path will be passed through to $EDITOR if doing a interactive rebase.",
+        group = "mutually-exclusive",
```

Comment:
```suggestion
        group = "mutually-exclusive",
        hidden = true,
```
Maybe hide this option from the help text?

Reply:
Sounds good 👍